django/db/models/fields/__init__.py

```diff
@@ -2397,9 +2397,11 @@ def as_sql(self, compiler, connection):
             tzname = timezone.get_current_timezone_name() if settings.USE_TZ else None
             sql, tz_params = connection.ops.datetime_extract_sql(self.lookup_name, sql, tzname)
             params.extend(tz_params)
-        else:
-            # DateField and TimeField.
+        elif isinstance(lhs_output_field, DateField):
             sql = connection.ops.date_extract_sql(self.lookup_name, sql)
+        else:
+            # TimeField.
+            sql = connection.ops.time_extract_sql(self.lookup_name, sql)
```

Comment:
Thanks :) I had matched what was there before, but if that's all the uses of DateTransform, that makes sense! Changed, and also docs updated to mention the new options.